frontend/vue/components/UserAccount/UserAccountLayout.vue

```diff
@@ -0,0 +1,196 @@
+        PROGRESS SECTION
+      </div>
+      <div v-if="activeSection === sectionList[1].hash">
+        GROUPS SECTION
+      </div>
+      <div v-if="activeSection === sectionList[2].hash">
+        BADGES SECTION
+      </div>
+      <div v-if="activeSection === sectionList[3].hash">
+        PRIVACY SECTION
+      </div>
+    </section>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue-demi'
+import AppLink from '../common/AppLink.vue'
+
+export default defineComponent({
+  name: 'UserAccountLayout',
+  components: {
+    AppLink
+  },
+  props: {
+    userName: { type: String, required: false, default: 'Unknown' },
+    userRole: { type: String, required: false, default: 'Unknown' }
+  },
+  data () {
+    return {
+      activeSection: '',
+      sectionList: [
+        {
+          displayName: this.$translate('My Learning'),
+          hash: '#MyLearning'
+        },
+        {
+          displayName: this.$translate('Groups'),
+          hash: '#Groups'
+        },
+        {
+          displayName: this.$translate('Badges'),
+          hash: '#Badges'
+        },
+        {
+          displayName: this.$translate('Privacy'),
+          hash: '#Privacy'
+        }
+      ]
+    }
+  },
+  mounted () {
+    this.activeSection = window.location.hash || this.sectionList[0].hash
+    window?.addEventListener('hashchange', () => {
+      this.activeSection = window.location.hash
+    }, false)
+  },
+  methods: {
+    switchPanel (event: any) {
+      window.location.hash = event.detail.item.value
+    }
+  }
+})
+</script>
+
+<style lang="scss" scoped>
+@import '../../../../node_modules/carbon-components/scss/globals/scss/layout';
+@import '../../../../node_modules/carbon-components/scss/globals/scss/typography';
+@import '~/../scss/variables/colors.scss';
+@import '~/../scss/variables/fonts.scss';
+@import '~/../scss/variables/mq.scss';
+
+.user-account {
+  display: grid;
+  grid-template-columns: 17rem 1fr;
+  height: calc(100vh - 60px);
+  grid-template-areas: "navigation section";
+
+  @include mq($from: medium, $until: large) {
+    grid-template-columns: 12rem 1fr;
+  }
+  @include mq($until: medium) {
+    grid-template-areas: "section";
+    grid-template-columns: 1fr;
+    grid-template-rows: min-content 1fr;
+  }
+
+  &__section-dropdown {
+    --cds-field-01: #{$background-color-white};
+    --cds-hover-ui: #{$background-color-lighter};
+    --cds-ui-04: #{$border-color-tertiary};
+    --cds-ui-01: #{$background-color-white};
+
+    @include mq($from: medium) {
+      display: none;
+    }
+
+    &__item {
+      --cds-hover-selected-ui: #{$background-color-lighter};
+      --cds-selected-ui: #{$background-color-white};
+    }
+  }
+
+  &__section-nav {
+    grid-area: "navigation";
+    display: flex;
+    @include mq($until: medium) {
+      display: none;
+    }
+    flex-direction: column;
+    padding: $spacing-07;
+
+    background-color: $background-color-light;
```

Comment:
I think this might be reordered to be easier to read:

```suggestion
    background-color: $background-color-light;
    grid-area: "navigation";
    display: flex;
    flex-direction: column;
    padding: $spacing-07;

    @include mq($until: medium) {
      display: none;
    }
```